File: frontend/src/app/models/feedback.ts
export interface Feedback {
  id: number;
  eventId: number;
  userId: number;
  rating: number;
  comment: string;
  status: 'PENDING' | 'APPROVED' | 'REJECTED' | 'FLAGGED';
  moderatedBy?: number;
  moderationNote?: string;
  flaggedReason?: string;
  createdAt: string;
  updatedAt?: string;
  username?: string;
  eventTitle?: string;
}

export interface FeedbackRequest {
  eventId: number;
  userId: number;
  rating: number;
  comment: string;
}

export interface FeedbackStats {
  eventId: number;
  averageRating: number;
  totalFeedbacks: number;
  ratingDistribution: { [key: number]: number };
}

File: microservices/feedback-service/src/main/java/com/example/feedbackservice/entity/Feedback.java
package com.example.feedbackservice.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "feedbacks")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Feedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long eventId;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private int rating;

    @Column(length = 2000)
    private String comment;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private FeedbackStatus status = FeedbackStatus.PENDING;

    private Long moderatedBy;

    @Column(length = 1000)
    private String moderationNote;

    @Column(length = 500)
    private String flaggedReason;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
